docs/pull-modes.md

```diff
@@ -1,14 +1,36 @@
-## Pull modes of soci-snapshotter
+# Pull modes of soci-snapshotter
 
-Soci-snapshotter is a remote snapshotter. It is able to lazily load the contents of a container image when a *SOCI index* is present in the remote registry. If a SOCI index is not found, it will download and uncompress the layers at launch time, just like the default snapshotter does. SOCI indices can also be "sparse", meaning that any individual layer may not be indexed. In that case, that layer will be downloaded at launch time, while the indexed layers will be lazily loaded.
-Every layer which is being lazily loaded gets mounted as a fuse mountpoint. Every other layer gets mounted as an overlay layer.
-During the pull phase there can be the following cases:
-1. For lazy loading the snapshotter expects the user to pass `--soci-index-digest`, which should be the sha256 of the SOCI index manifest. If it’s not passed, the snapshotter will download and decompress the layers just as the default snapshotter does. It will then print `unable to find SOCI index digest for %v: the container image will be pulled in non-SOCI mode`. It will print `using SOCI index digest: %v` otherwise.
-2. If 1 goes well, the launch is a SOCI use case. On the first layer mount, there will be an attempt to download and parse the SOCI manifest. If this doesn’t go well, there will be the following error in the log: `unable to fetch SOCI artifacts: `, which will indicate that the container image will not be lazily loaded.
-3. If 1 and 2 go well. There can be a case, where there’s no zTOC for a specific layer. In this case, there will be an error with the text `{"error":"failed to resolve layer` which will indicate that the specific layer will be synchronously downloaded at launch time.
-4. With debug logging enabled, you can see an entry in the logs for each layer. `checking mount point` indicates that the layer will be lazily loaded. `layer is normal snapshot(overlayfs)` indicates that it will not be lazily loaded.
+soci-snapshotter is a remote snapshotter. It is able to lazily load the contents
+of a container image when a *SOCI index* is present in the remote registry. If
+a SOCI index is not found, it will download and uncompress the layers at launch
+time, just like the default snapshotter does.
 
-```
+SOCI indices can also be "sparse", meaning that any individual layer may not be
+indexed. In that case, that layer will be downloaded at launch time, while the
+indexed layers will be lazily loaded.
+
+A layer will be mounted as a fuse mountpoint if it's being lazily loaded, or as
+a normal overlay layer if it's not.
+
+When you pull an image, there can be the following cases:
+
+1. For lazy loading the snapshotter expects the user to pass `--soci-index-digest`,
```

Comment:
I feel like I saw a comment somewhere about this, but now I can't find it.

`--soci-index-digest` is optional now.

Reply:
yeah I just did some formatting in this PR. I did open an issue to clarify this https://github.com/awslabs/soci-snapshotter/issues/488. I can update it altogether later in this PR.

Reply:
update this PR to highlight the flag is optional and referral is used if not present. But this doc still has other issues. The detail is in #488